File: src/ipc/vault.ts
import { invoke } from "@tauri-apps/api/core";

export const setVaultPassword = (deviceId: string, password: string): Promise<void> =>
  invoke("vault_set_password", { deviceId, password });

export const getVaultPassword = (deviceId: string): Promise<string | null> =>
  invoke("vault_get_password", { deviceId });

export const deleteVaultPassword = (deviceId: string): Promise<void> =>
  invoke("vault_delete_password", { deviceId });

File: src-tauri/src/commands/vault.rs
use crate::core::vault;

#[tauri::command]
pub fn vault_set_password(device_id: String, password: String) -> Result<(), String> {
    vault::set_password(&device_id, &password).map_err(|e| e.to_string())
}

#[tauri::command]
pub fn vault_get_password(device_id: String) -> Result<Option<String>, String> {
    vault::get_password(&device_id).map_err(|e| e.to_string())
}

#[tauri::command]
pub fn vault_delete_password(device_id: String) -> Result<(), String> {
    vault::delete_password(&device_id).map_err(|e| e.to_string())
}
